packages/firestore-types/index.d.ts

```diff
@@ -166,7 +177,13 @@ export class WriteBatch {
 
   set<T>(
     documentRef: DocumentReference<T>,
-    data: T,
+    data: Partial<T>,
+    options: SetOptions
+  ): WriteBatch;
+  set<T>(documentRef: DocumentReference<T>, data: T): WriteBatch;
+  set<T>(
+    documentRef: DocumentReference<T>,
+    data: T | Partial<T>,
     options?: SetOptions
   ): WriteBatch;
```

Comment:
This catch-all override should not be needed in the type file.

Reply:
removed.